rgw: aws sync module fix

Leeshine opened the pull request:
- The first commit is to decrease the log level in ```RGWCoroutinesStack::unwind```
- The second commit is to add a sub-class of ```RGWRESTConn``` called ```S3RESTConn``` that do not
contain the rgwx-zonegroup param
- The third commit is to add populate_zonegroup() and populate_uid()
- The fourth commit is to modify some testcase in in src/test/rgw

Leeshine committed:
rgw: decrease the log level in RGWCoroutinesStack::unwind

Signed-off-by: lvshanchun <lvshanchun@gmail.com>

yehudasa commented on src/rgw/rgw_rest_conn.h:
```diff
@@ -90,6 +90,8 @@ class RGWRESTConn
   }
   size_t get_endpoint_count() const { return endpoints.size(); }
 
+  virtual void populate_params(param_vec_t& params, const rgw_user *uid, const string& zonegroup);
```
@Leeshine `populate_params()` is too generic, let's in addition have `populate_zonegroup()` and use that one instead.

yehudasa reviewed (commented):
@Leeshine see my comment

Leeshine commented:
ping @yehudasa

Leeshine committed:
rgw: implement S3RESTConn to send request to S3
add a sub-class of RGWRESTConn called S3RESTConn that do not
populate zonegroup-related params, it shall only used for send request
to S3-like interface.

Signed-off-by: lvshanchun <lvshanchun@gmail.com>

Leeshine committed:
rgw: add populate_zonegroup() and populate_uid() in RGWRESTConn

populate_params() is too generic, add populate_zonegroup() and
populate_uid() in RGWRESTConn, then have populate_params() call them.

Signed-off-by: lvshanchun <lvshanchun@gmail.com>

Leeshine committed:
rgw: modify testcase in src/test/rgw

Signed-off-by: lvshanchun <lvshanchun@gmail.com>

yehudasa commented:
@Leeshine instead of replacing `populate_params()` with `populate_zonegroup()` have both interfaces. `populate_zonegroup()` can be private. You can also have one for setting the uid. Have `populate_params()` call them.

Leeshine commented:
ping @yehudasa
